assets/js/app.coffee

```diff
@@ -70,6 +71,109 @@ class Choosealicense
   # Post-copy user feedback callback
   clipboardComplete: (client, args) ->
     @textContent = "Copied!"
+  
+  # Initializes the repository suggestion feature
+  initLicenseSuggestion: ->
+    inputEl = $("#repository-url")
+    licenseId = inputEl.attr("data-license-id")
+    statusIndicator = $(".status-indicator")
+    new LicenseSuggestion(inputEl, licenseId, statusIndicator)
 
+class LicenseSuggestion
+  constructor: (@inputEl, @licenseId, @statusIndicator) ->
+    @setupTooltips()
+    @bindEventHandlers()
+  
+  # Initializes tooltips on the input element
+  setupTooltips: =>
+    @inputEl.qtip
+      content:
+        text: false
+        title:
+          text: "message"
+      show: false
+      hide: false
+      position:
+        my: "top center"
+        at: "bottom center"
+      style:
+        classes: "qtip-shadow"
+  
+  # Main event handlers for user input
+  bindEventHandlers: =>    
+    @inputEl.on "input", (event) =>
+      @setStatus ""
+    .on "keyup", (event) =>            
+      if event.keyCode == 13 and event.target.value
+        # Validate the user input first
+        try
+          repositoryFullName = @parseUserInput event.target.value
+        catch
+          @setStatus "Error", "Invalid URL."
+          return
+        
+        @setStatus "Fetching"        
+        @fetchInfoFromGithubAPI repositoryFullName, (err, repositoryInfo=null) =>
+          if (err)
+            @setStatus "Error", err.message
+            return
+          if repositoryInfo.license # The repository already has a license
+            license = repositoryInfo.license
+            @setStatus "Error", @repositoryLicense repositoryFullName, license
+          else # The repository is unlicensed
+            window.open "https://github.com/#{repositoryFullName}/community/license/new?template=#{@licenseId}"                
```

Comment:
This seems to trigger both Firefox and Chrome popup blockers. Would `location.href` work instead?

Reply:
Good catch!